django/db/backends/oracle/operations.py

```diff
@@ -636,4 +636,6 @@ def conditional_expression_supported_in_where_clause(self, expression):
             return True
         if isinstance(expression, ExpressionWrapper) and isinstance(expression.expression, Q):
             return True
+        if isinstance(expression, RawSQL) and expression.conditional:
+            return True
```

Comment:
This patch assumes that conditional `RawSQL`'s expressions are filters, which sounds fine for this feature. Nevertheless it breaks uncommon use cases, e.g.
```python
Company.objects.filter(RawSQL('1', (), output_field=models.BooleanField())).count()
```
I think we can document this caveat on Oracle, e.g.
```
Conditional ``RawSQL`` expressions are assumed to be filters on Oracle, because
other boolean expressions are not supported in the ``WHERE`` clause by this
database.
```

Reply:
I decided that this usage is so niche that we don't need to document it (at least for now).